### PR TITLE
feat: generate dynamic number of random columns

### PR DESCRIPTION
**Pull request by Macbeth98**
# Description
- Generates Dynamic Random number of Columns

#### To generate 100 columns per record:

```
go run main.go ingest esbulk -n 1 -g benchmark -d http://localhost:8081/elastic -t 1000 -p 1 --enableVariableNumColumns true
```

#### To generate `N` columns per record:

```
go run main.go ingest esbulk -n 1 -g benchmark -d http://localhost:8081/elastic -t 1000 -p 1 --enableVariableNumColumns true --maxColumns 200
```

#### To generate variable number of random columns per record with a max of `N1` columns and a min of `N2` columns per record:

```
go run main.go ingest esbulk -n 1 -g benchmark -d http://localhost:8081/elastic -t 1000 -p 1 --enableVariableNumColumns true --maxColumns 200 --minColumns 50
```

```
go run main.go ingest esbulk -n 1 -g benchmark -d http://localhost:8081/elastic -t 100_000 -p 1 --enableVariableNumColumns true --minColumns 70
```
**NOTE**: In the above case the default max value is 100 columns.

#### Defaults:
- variableColumns: `false`. Should be enabled for any of these random Operations
- numColumns: 100
- minColumns: 0

# Testing
- Tested by ingesting `100K` records with these configs and 100 columns on average per record.

# Checklist:

- [x] I have self-reviewed this PR.
- [x] I have removed all print-debugging and commented-out code that should not be merged.
- [x] I have added sufficient comments in my code, particularly in hard-to-understand areas.
- [x] I have formatted the code, if applicable. For Go, I have run `goimports -w .`.

### CODE REVIEW

File: tools/sigclient/README.md [hunk 1]
```diff
@@ -50,6 +50,10 @@ Options:
   -t, --totalEvents int      Total number of events to send (default 1000000)
   -u, --uniqueness int       Cardinality (uniqueness) of the data (default 2000000)
   -e, --eventsPerDay uint    Number of events to ingest per day. If set, the ingestion mode will be assumed to be continuous.
+      --randomColumns bool   Set this to true to generate random number of columns. By Default this will generate 200 Columns.
```
Comment (nkunal) [inline tools/sigclient/README.md:53]: default should be 100

**Review by nkunal — COMMENTED** [review]
overall looks good. Whatever you have in the PR description, add that in the readme, so it is easy to find on how to use this new config

**Review by nkunal — DISMISSED** [review]

File: tools/sigclient/README.md [hunk 2]
```diff
@@ -26,6 +26,9 @@ Options:
   -c  continuous             If true, ignores -t and will continuously send docs to the destination
   -o  --outputFile           filepath to output the query response Time results in csv format.
   -t  --runResponseTime      If true, then the queries will be run and the response time will be recorded in the given/default outputFile in CSV Format.
+      --variableColums bool  Set this to true to generate variable number of columns per record. Each record will have a random number of columns between minColumns and maxColumns
```
Comment (nkunal) [inline tools/sigclient/README.md:29]: typo in name

File: tools/sigclient/README.md [hunk 3]
```diff
@@ -26,6 +26,9 @@ Options:
...
+      --maxColumns  int      Maximum number of Columns to generate. By Default this value is set to 100.
+      --minColumns int       Minimum number of Columns to generate. Default value is 0.
```
Comment (nkunal) [inline tools/sigclient/README.md:31]: should the default be 0 ?

File: tools/sigclient/README.md [hunk 4]
```diff
@@ -34,6 +37,18 @@ Different Types of Readers:
 2. Dynamic User: Randomly Generates user events. These random events are generated using [gofakeit](github.com/brianvoe/gofakeit/v6).
 3. File: Reads a file line by line. Expects each line is a new json. Will loop over file if necessary
 
+To Ingest 200 columns per record:
+
+```bash
+go run main.go ingest esbulk -n 1 -g benchmark -d http://localhost:8081/elastic -t 1000 -p 1 --enableVariableNumColumns true --maxColumns 200
```
Comment (nkunal) [inline tools/sigclient/README.md:43]: ijn this case `minColumns` will default to 0, so would it generate between 0 and 200 num of columns per record ?

File: tools/sigclient/cmd/sigclient.go [hunk 5]
```diff
@@ -76,8 +88,11 @@ var esBulkCmd = &cobra.Command{
 		log.Infof("bearerToken : %+v\n", bearerToken)
 		log.Infof("generatorType : %+v. Add timestamp: %+v\n", generatorType, ts)
 		log.Infof("eventsPerDay : %+v\n", eventsPerDay)
+		if enableVariableNumColumns {
```
Comment (nkunal) [inline tools/sigclient/cmd/sigclient.go:91]: nit we should keep the format similar to other prints we have earlier. 
if they are not set , then should display the default value

Comment (nkunal) [inline tools/sigclient/cmd/sigclient.go:91]: min and max cols also need to be shown here

File: tools/sigclient/README.md (hunk 3 above)
Comment (Macbeth98) [reply tools/sigclient/README.md:31]: Change the default to `100`

File: tools/sigclient/README.md (hunk 4 above)
Comment (Macbeth98) [reply tools/sigclient/README.md:43]: If `minColumns` is 0, then that value will be set to the `maxColumns` value which is `200` in this case.

**Review by nkunal — APPROVED** [review]

**Pull request merged**